src/steps/internal.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { SomeRequired } from '../internal/types';
+import StatusIndicator from '../status-indicator/internal';
+import { StepsProps } from './interfaces';
+
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+
+type InternalStepsProps = SomeRequired<StepsProps, 'steps'> & InternalBaseComponentProps<HTMLDivElement>;
+
+const InternalStep = ({ status, statusIconAriaLabel, header, details }: StepsProps.Step) => {
+  return (
+    <li className={clsx(styles['step-container'], testUtilStyles['step-container'])}>
+      <div className={clsx(styles['step-status'], testUtilStyles['step-status'])}>
+        <StatusIndicator type={status} iconAriaLabel={statusIconAriaLabel} />
```

Comment:
It seems that only icon is being used from status indicator? If we only want to get the icon then why not use the icon component? 
Another question, do we expect the text to be the same color as the status icon? e.g. green when status is success?

Reply:
> Another question, do we expect the text to be the same color as the status icon? e.g. green when status is success?

That is really up to the consumer. There are use cases where consumers want to do that and others where they don't

Reply:
Let's discuss switching it to icon with Kashi as well. If it's an icon, we could even expose the other available icon names and custom icons through url or svg but there could be a specific reason why status indicator was chosen (which could be the point I brought up whether we expect text to be green, red, etc.)

Reply:
>  If we only want to get the icon then why not use the icon component?

I was looking into this and one issue is that the https://refresh.cloudscape.aws.dev/components/icon does not have Loading, so we would need to provide some custom logic for that. Because of that I would propose continuing with using the StatusIndicator. WDYT?

Reply:
Hmm true that is lacking but we can easily switch it with a spinner. Though it'd mean some API changes and implementation changes. I'd say you leave it as is right now and then we discuss with Kashi